src/Simplex/Chat.hs

```diff
@@ -1655,6 +1655,43 @@ processChatCommand' vr = \case
     case conn'_ of
       Just conn' -> pure $ CRConnectionIncognitoUpdated user conn'
       Nothing -> throwChatError CEConnectionIncognitoChangeProhibited
+  APISetConnectionUserId connId newUserId -> withUser $ \user@User {userId} -> do
+    cfg <- asks config
+    newUser <- privateGetUser newUserId
+    conn' <- withFastStore $ \db -> do
+      conn <- getPendingContactConnection db userId connId
+      newUserServers <- liftIO $ getServers db cfg newUser SPSMP
+      pure $ checkConnection conn newUserServers
+    case conn' of
+      (True, Just conn) -> do
+        -- Safe to update connection
+        withAgent $ \a -> changeConnectionUser a (aUserId user) (aConnId' conn) (aUserId newUser)
+        _ <- withFastStore' $ \db -> updatePCCUser db userId conn newUserId
+        pure $ CRConnectionUserChanged user conn newUser
+      (False, Just conn) -> do
+        -- Servers do not match, create new connection
+        subMode <- chatReadVar subscriptionMode
+        (newConnId, cReq) <- withAgent $ \a -> createConnection a (aUserId user) True SCMInvitation Nothing IKPQOn subMode
+        newConn <- withFastStore' $ \db -> do
+          deleteConnectionRecord db user connId
+          createDirectConnection db newUser newConnId cReq ConnNew Nothing subMode initialChatVersion PQSupportOn
+        withAgent $ \a -> deleteConnectionAsync a False (aConnId' conn)
+        pure $ CRInvitation user cReq newConn
+      _ -> throwChatError CEInvalidConnReq
+    where
+      getServers :: (ProtocolTypeI p, UserProtocol p) => DB.Connection -> ChatConfig -> User -> SProtocolType p -> IO (NonEmpty (ProtocolServer p))
```

Comment:
not necessary to define general function as it's only used for smp

Reply:
would be smth like this:

`L.map (\ServerCfg {server} -> protoServer server) . useServers cfg SPXFTP <$> withStore' (`getProtocolServers` user)`

Reply:
we need to move this logic of retrieving user servers into some separate function already.. I was about to do this refactoring previous time it got repeated again with minor variation. There's 4 or 5 places already that do the same simple thing of reading servers from db and substituting with default if empty, that do it slightly differently every time to accommodate to downstream type instead of mapping after, and each time you have to figure out what's the logic there. not proposing to do it in this pr.